implementations/rust/ockam/ockam/src/entity/local.rs

```diff
@@ -0,0 +1,219 @@
+use ockam_node::{Cancel, Context};
+use ockam_vault_core::{PublicKey, Secret};
+use ockam_vault_sync_core::Vault;
+
+use ockam_core::hashbrown::HashMap;
+
+pub struct LocalEntity {
+    pub ctx: Context,
+    vault: Address,
+    entity: Entity<ProfileSync>,
+    secure_channels: HashMap<Address, Route>,
+}
+
+impl LocalEntity {
+    pub async fn create<A: Into<Address>>(node_ctx: &Context, address: A) -> Result<LocalEntity> {
+        let ctx = node_ctx.new_context(address).await?;
+
+        let vault = Vault::create(&ctx)?;
+        let default_profile = Profile::create(&ctx, &vault).await?;
+        let entity = Entity::new(default_profile);
+
+        Ok(LocalEntity {
+            ctx,
+            vault,
+            entity,
+            secure_channels: Default::default(),
+        })
+    }
+
+    /// Create with a given worker
+    pub async fn create_with_worker<A, M, W>(
+        ctx: &Context,
+        address: A,
+        worker: W,
+    ) -> Result<LocalEntity>
+    where
+        A: Into<AddressSet>,
+        M: Message + Send + 'static,
+        W: Worker<Context = Context, Message = M>,
+    {
+        #[allow(clippy::needless_lifetimes)]
+        let local = LocalEntity::create(ctx, Address::random(0)).await?;
+
+        ctx.start_worker(address, worker).await?;
+
+        Ok(local)
+    }
+
+    pub async fn secure_channel_listen_on_address<A: Into<Address>>(
+        &mut self,
+        address: A,
+    ) -> Result<()> {
+        self.entity
+            .create_secure_channel_listener(&self.ctx, address.into(), &self.vault)
+            .await
+    }
+
+    pub fn secure_channel_address(&self) -> String {
+        "secure_channel_listener".to_string()
+    }
+
+    pub async fn create_secure_channel_listener(
+        &mut self,
+        secure_channel_address: &str,
+    ) -> Result<()> {
+        self.secure_channel_listen_on_address(secure_channel_address)
+            .await
+    }
+
+    pub async fn create_secure_channel<R: Into<Route>>(&mut self, route: R) -> Result<Address> {
+        let route = route.into();
+        let channel = self
+            .entity
+            .create_secure_channel(&self.ctx, route.clone(), &self.vault)
+            .await?;
+
+        self.secure_channels.insert(channel.clone(), route);
+
+        Ok(channel)
+    }
+
+    pub fn list_secure_channels(&self) -> Result<Vec<(&Address, &Route)>> {
+        Ok(self.secure_channels.iter().collect())
+    }
+
+    pub async fn forward<A: Into<Address>>(
+        &mut self,
+        remote_entity: RemoteEntity,
+        service_address: A,
+    ) -> Result<RemoteForwarderInfo> {
+        let address = remote_entity.route.next().unwrap().to_string();
+        let address = address.strip_prefix("0#").unwrap(); // TODO how can we clean this up?
+        RemoteForwarder::create(&self.ctx, address, service_address.into()).await
+    }
+
+    pub async fn send<R, M>(&self, route: R, msg: M) -> Result<()>
```

Comment:
Not sure why we need send&receive here

Reply:
I'll remove - this was from before when moving the app Context instead of using `new_context`